src/mapgen/mapgen_valleys.cpp

```diff
@@ -529,48 +425,37 @@ int MapgenValleys::generateTerrain()
 			index_3d += ystride;
 		}
 
-		if (heightmap[index_2d] == -MAX_MAP_GENERATION_LIMIT) {
-			s16 surface_y_int = myround(surface_y);
-
-			if (surface_y_int > node_max.Y + 1 ||
-					surface_y_int < node_min.Y - 1) {
-				// If surface_y is outside the chunk, it's good enough
-				heightmap[index_2d] = surface_y_int;
-			} else {
-				// If the ground is outside of this chunk, but surface_y is
-				// within the chunk, give a value outside.
-				heightmap[index_2d] = node_min.Y - 2;
-			}
-		}
-
 		// Optionally increase humidity around rivers
 		if (spflags & MGVALLEYS_HUMID_RIVERS) {
+			// Compensate to avoid increasing average humidity
+			m_bgen->humidmap[index_2d] *= 0.8f;
 			// Ground height ignoring riverbeds
-			float t_alt = std::fmax(noise_rivers->result[index_2d],
-				(float)heightmap[index_2d]);
-			float water_depth = (t_alt - river_y) / humidity_dropoff;
+			float t_alt = std::fmax(base, (float)column_max_y);
+			float water_depth = (t_alt - base) / 4.0f;
```

Comment:
Please add a short comment about what `/ 4.0` is (humidity dropoff?)

Reply:
I don't see a need for a comment, this is just an arbitrary tuning value.

Reply:
Oh actually, rivers are about 4 nodes deep, so this is water_depth 0-1. Still don't see a need for a comment.